pkg/compiler/analysis.go

```diff
@@ -83,6 +90,11 @@ func (c *codegen) traverseGlobals() (int, bool) {
 			// encountered after will be recognized as globals.
 			c.scope = nil
 		})
+		// store auxiliary variables after all others.
+		if hasDefer {
+			c.exceptionIndex = len(c.globals)
+			c.globals["<exception>"] = c.exceptionIndex
```

Comment:
But still what prevents us from moving it into the local context?

Reply:
We store exception in the caller and `recover()` is executed in a callee, As we don't support closures, they share only global variables.

Reply:
What prevents us from passing it as a parameter to the deferred function?

Reply:
We can call regular functions in `defer`, and it is impossible to determine which context we are executed in (with the extra parameter or not).
But if we restrict ourselves to only function literals, it shouldn't be hard.